experimental/dds/tree2/src/test/shared-tree/editing.spec.ts

```diff
@@ -120,73 +120,78 @@ describe("Editing", () => {
 
-			const sequenced = sequencer.sequence([addW, delY]);
-			tree1.receive(sequenced);
-			tree2.receive(sequenced);
-
-			expectJsonTree([tree1, tree2], ["w", "x"]);
+			expectJsonTree([addW, delY], ["w", "x"]);
 		});
 
 		it("inserts that concurrently target the same insertion point do not interleave their contents", () => {
-			const sequencer = new Sequencer();
-			const tree1 = TestTree.fromJson([]);
-			const tree2 = tree1.fork();
-			const tree3 = tree1.fork();
-			const tree4 = tree1.fork();
+			const tree = makeTreeFromJson([]);
+			const abc = tree.fork();
+			const rst = tree.fork();
+			const xyz = tree.fork();
 
-			const abc = insert(tree1, 0, "a", "b", "c");
-			const rst = insert(tree2, 0, "r", "s", "t");
-			const xyz = insert(tree3, 0, "x", "y", "z");
+			insert(abc, 0, "a", "b", "c");
+			insert(rst, 0, "r", "s", "t");
+			insert(xyz, 0, "x", "y", "z");
 
-			const sequenced = sequencer.sequence([xyz, rst, abc]);
-			tree1.receive(sequenced);
-			tree2.receive(sequenced);
-			tree3.receive(sequenced);
-			tree4.receive(sequenced);
+			tree.merge(xyz);
+			tree.merge(rst);
+			tree.merge(abc);
 
-			expectJsonTree(
-				[tree1, tree2, tree3, tree4],
-				["a", "b", "c", "r", "s", "t", "x", "y", "z"],
-			);
+			xyz.rebaseOnto(tree);
+			rst.rebaseOnto(tree);
+			abc.rebaseOnto(tree);
+
+			expectJsonTree([tree, abc, rst, xyz], ["a", "b", "c", "r", "s", "t", "x", "y", "z"]);
 		});
 
 		it("merge-left tie-breaking does not interleave concurrent left to right inserts", () => {
-			const sequencer = new Sequencer();
-			const tree1 = TestTree.fromJson([]);
-			const tree2 = tree1.fork();
-			const tree3 = tree1.fork();
-			const tree4 = tree1.fork();
-
-			const a = insert(tree1, 0, "a");
-			const b = insert(tree1, 1, "b");
-			const c = insert(tree1, 2, "c");
-
-			const r = insert(tree2, 0, "r");
-			const s = insert(tree2, 1, "s");
-			const t = insert(tree2, 2, "t");
-
-			const x = insert(tree3, 0, "x");
-			const y = insert(tree3, 1, "y");
-			const z = insert(tree3, 2, "z");
-
-			const sequenced = sequencer.sequence([x, r, a, s, b, y, c, z, t]);
-			tree1.receive(sequenced);
-			tree2.receive(sequenced);
-			tree3.receive(sequenced);
-			tree4.receive(sequenced);
-
-			expectJsonTree(
-				[tree1, tree2, tree3, tree4],
-				["a", "b", "c", "r", "s", "t", "x", "y", "z"],
-			);
+			const tree = makeTreeFromJson([]);
+			const a = tree.fork();
+			const r = tree.fork();
+			const x = tree.fork();
+
+			insert(a, 0, "a");
+			const b = a.fork();
+			insert(b, 1, "b");
+			const c = b.fork();
+			insert(c, 2, "c");
+
+			insert(r, 0, "r");
+			const s = r.fork();
+			insert(s, 1, "s");
+			const t = s.fork();
+			insert(s, 2, "t");
+
+			insert(x, 0, "x");
+			const y = x.fork();
+			insert(y, 1, "y");
+			const z = y.fork();
+			insert(z, 2, "z");
+
+			tree.merge(x);
+			tree.merge(r);
+			tree.merge(a);
+			tree.merge(s);
+			tree.merge(b);
+			tree.merge(y);
+			tree.merge(c);
+			tree.merge(z);
```

Comment:
Note an issue with this PR, but with the new branch API I see no way that branches reach their end of life. 

This PR depends on the branches loving past the merge, but performance and memory use (ex: shrinking the effective collab window including local branches) critically depends on branches not being leaked, and with merge not implicitly ending their life, and not clear life ending API, its unclear how this is supposed to work.

An example API that could solve this problem would be an optional second parameter to merge, a Boolean to request the source branch stay alive, defaulting to false, as well as an API to delete a branch, and docs on forks making these requirements clear.

Reply:
@noencke FYI

Reply:
Cool. Am I understanding right that these tests don't need to call `dispose()`? Does that all get collected when we exist the test scope?

Reply:
Right - there's nothing magic happening that requires we dispose (like dangling singletons or something). Dispose simply marks the branch as "no longer usable", and then `EditManager` listens for a dispose event from all known branches and evicts old data from the trunk. This only matters in long running scenarios, like a real app. In your case, you aren't even using an `EditManager` for some tests (e.g. if you create your root view using `createSharedTreeView()`), so even more so there is no need to dispose. It could be argued that it is good practice to dispose, but for most tests it doesn't matter.